server/pulp/server/controllers/repository.py

```diff
@@ -186,20 +194,39 @@ def has_all_units_downloaded(repository):
     :return: True if completely downloaded
     :rtype: bool
     """
-    types = {}
-    q_set = model.RepositoryContentUnit.objects(repo_id=repository.repo_id)
-    for association in q_set.only('unit_id', 'unit_type_id'):
-        unit_ids = types.setdefault(association.unit_type_id, [])
-        unit_ids.append(association.unit_id)
-    for type_id, unit_ids in types.items():
-        _model = plugin_api.get_unit_model_by_id(type_id)
-        for page in paginate(unit_ids):
-            q_set = _model.objects(id__in=page, downloaded=False)
-            if q_set.count():
-                return False
+    for qs in _get_mongoengine_repo_querysets(str(repository.repo_id)):
+        if qs(downloaded=False).count():
+            return False
     return True
 
 
+def _get_mongoengine_repo_querysets(repo_id, repo_content_unit_q=None):
```

Comment:
>     # Filter any non-MongoEngine content types.
>     return [m for m in unit_models if m is not None]

If you want to filter, I think you should `filter`. `filter(None, unit_models)` will return a list of truthies. That's probably(?) good enough for model instances, but if explicit comparison to None is needed `filter(lambda model: model is not None, unit_models)`.